internal/provider/providers/porkbun/provider.go

```diff
@@ -141,6 +142,7 @@ func (p *Provider) Update(ctx context.Context, client *http.Client, ip netip.Add
 
 	for _, record := range records {
 		err = p.updateRecord(ctx, client, recordType, p.owner, ipStr, record.ID)
+		time.Sleep(time.Second)
```

Comment:
I ran into this too.

One thing, is that if we're getting rate-limited, likely adding the same sleep delay here won't solely fix the underlying issue. This fixes it for lots of subdomains under the same hostname, but if you have DDNS-updater setup to do lots of different hostnames, you can also run into rate-limiting (as I have encountered).

@qdm12 any thoughts on how you'd want this fixed? Maybe wrap the HTTP client to have a general http request rate limit, either per host, or globally. Would that make sense to you? In my mind, you'd ideally be able to specify a max requests per period for a given domain, and the http client would automatically handle queuing and staying within that request limit.

Reply:
This also might be pretty unique to porkbun and we'd just want to setup a porkbun ratelimiter